lib/mix/lib/mix/tasks/test.ex

```diff
@@ -569,6 +574,25 @@ defmodule Mix.Tasks.Test do
     end
   end
 
+  defp normalize_max_failures(opts) do
+    case Keyword.fetch(opts, :max_failures) do
+      {:ok, value} when value in ["infinity", ":infinity"] ->
+        Keyword.put(opts, :max_failures, :infinity)
```

Comment:
I don't think we need to support this. The equivalent of setting it to infinity is not setting it at all.